examples/SDG1.ipynb

```diff
@@ -0,0 +1,1336 @@
+        "\n",
+        "We know that there should be 4 cluster (4 topics) since we specified this in prompt: vehicle, electronics, clothing, food. However in general for our data, we do not know the number of clusters that exist. Therefore we will use the elbow method. Here we iterate through a range of different K's, each time storing the inertia. The inertia measures the sum of the squared distances between each point in a cluster and the centroid of that cluster thus telling us how well-separated and dense each cluster is. If we plot K against the inertia, we are able to see how the inertia drops and where the drop in inertia is least rapid (often making an elbow shape) we can set our optimal number of clusters."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "1BxwPTkpGzu8"
+      },
+      "source": [
+        "First let's store our data into a pandas dataframe for ease of analysis\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 81,
+      "metadata": {
+        "id": "XcPBzORtKWv6"
+      },
+      "outputs": [],
+      "source": [
+        "data = {\n",
+        "    'Product': products,\n",
+        "    'Category': categories,\n",
+        "    'Description': descriptions\n",
+        "}\n",
+        "\n",
+        "df = pd.DataFrame(data)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "HQbg6r37KjG0"
+      },
+      "source": [
+        "Next let us embed our data as the embeddings is what we will cluster since they should be close to each other in vector space if they are similar."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 82,
+      "metadata": {
+        "id": "l8M7MX-1Jctr"
+      },
+      "outputs": [],
+      "source": [
+        "def get_embedding(text, model=\"text-embedding-3-small\"):\n",
+        "    text = text.replace(\"\\n\", \" \")\n",
+        "\n",
+        "    response = client.embeddings.create(input=[text], model=model)\n",
+        "\n",
+        "    return response.data[0].embedding\n",
+        "\n",
+        "embedding_model = \"text-embedding-3-small\"\n",
+        "df[\"embedding\"] = df.Category.apply(lambda x: get_embedding(x, model=embedding_model))\n",
+        "\n",
+        "matrix = np.vstack(df.embedding.values)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ZcdZscBNKy0F"
+      },
+      "source": [
+        "Now we perform the elbow method. This will output a chart for us in which we have to visually tell where the optimal cluster point is. We can see below that we see a gradual decrease of inertia rather than a sharp elbow but the point of steepest decrease appears to occur around 3, 4 or 5 clusters which lines up with our expectations given our prompt. \n",
```

Comment:
can we perhaps add more info/background into the elbow method?

Reply:
More than this paragraph feels like too much depth for this cookbook. I added a link to read more about it elsewhere for more info/background.